osquery/events/linux/auditdnetlink.cpp

```diff
@@ -392,54 +393,41 @@ bool AuditdNetlinkReader::configureAuditService() noexcept {
     }
   }
 
+  audit_rule_data rule = {};
+
   // Attempt to add each one of the rules we collected
   for (int syscall_number : monitored_syscall_list_) {
-    audit_rule_data rule = {};
     audit_rule_syscall_data(&rule, syscall_number);
-
-    // clang-format off
-    int rule_add_error = audit_add_rule_data(audit_netlink_handle_, &rule,
-      // We want to be notified when we exit from the syscall
-      AUDIT_FILTER_EXIT,
-
-      // Always audit this syscall event
-      AUDIT_ALWAYS
-    );
-    // clang-format on
-
-    // When exiting, don't remove the rules that were already installed, unless
-    // we have been asked to
-    if (rule_add_error >= 0) {
-      if (FLAGS_audit_debug) {
-        VLOG(1) << "Audit rule installed for syscall " << syscall_number;
-      }
-
-      installed_rule_list_.push_back(rule);
-      continue;
-    }
-
     if (FLAGS_audit_debug) {
-      VLOG(1) << "Audit rule for syscall " << syscall_number
-              << " could not be installed: " << (-errno);
+      VLOG(1) << "Audit rule queued for syscall " << syscall_number;
     }
+  }
 
-    if (FLAGS_audit_force_unconfigure) {
-      installed_rule_list_.push_back(rule);
-    }
+  // clang-format off
+  int rule_add_error = audit_add_rule_data(audit_netlink_handle_, &rule,
+    // We want to be notified when we exit from the syscall
+    AUDIT_FILTER_EXIT,
 
-    rule_add_error = -rule_add_error;
+    // Always audit this syscall event
+    AUDIT_ALWAYS
+  );
+  // clang-format on
 
-    if (rule_add_error != EEXIST) {
-      VLOG(1) << "The following syscall number could not be added to the audit "
-                 "service rules: "
-              << syscall_number << ". Some of the auditd "
-              << "table may not work properly (process_events, "
-              << "socket_events, process_file_events, user_events)";
-    }
+  if (rule_add_error < 0) {
+    LOG(ERROR) << "Failed to install the audit rule due to one more more "
+               << "syscalls due to error "
+               << audit_errno_to_name(rule_add_error)
+               << ", Audit-based tables may not function as expected";
```

Comment:
Calling `audit_errno_to_name()` is a great addition! There's a small error in the message, otherwise it looks great!

Reply:
Ahh!, the typos.. I'll fix that.